src/array.jl

```diff
@@ -112,3 +117,27 @@ function tomatrix(M::ITensor; reverse::Bool = true)
   return ITensors.array(permute(Mmat, c', c))
 end
 
+"""
+    toitensor(M::Array, sites::Vector{<:Index})
+
+Generate an itensor from a dense vector of matrix
+"""
+function toitensor(M::Array, sites::Vector{<:Index})
+  d = M isa Vector ?  length(M) : size(M,1) 
+  n = length(sites)
+  totaldim = 1.0
+  for j in 1:n
+    totaldim *= dim(sites[j])
+  end
+  @assert totaldim == d
+  
+  sites = reverse(sites)
+  if M isa Vector
+    M = reshape(M, (repeat([2],n)...))
+    return itensor(M, sites)
+  else
+    reshape(M, (repeat([2],2*n)...))
+    return itensor(M, prime.(sites)..., ITensors.dag.(sites)...) 
```

Comment:
@mtfishman double check for QN

Reply:
This should be fine for QNs. The `itensor` constructor automatically drops zeros to try to make it sparse. The main issue is if they are not exactly zero, in which case `itensor` has a tolerance keyword argument which we will probably want to pass.

Reply:
It seems better to define this in two functions as `toitensor(::Vector, ...)` and `toitensor(::Matrix, ...)`. Also I'm not totally sure of the utility of this beyond the standard `itensor(::Array, ...)` function, I imagine you have a particular use case for this?

Reply:
I'm still not quite sure this needs to be a function, it is pretty simple with `itensor` directly:
```julia
julia> v = randn(2^3);

julia> s = siteinds("Qubit", 3);

julia> itensor(v, s)
ITensor ord=3 (dim=2|id=535|"Qubit,Site,n=1") (dim=2|id=101|"Qubit,Site,n=2") (dim=2|id=268|"Qubit,Site,n=3")
ITensors.NDTensors.Dense{Float64, Vector{Float64}}

julia> m = randn(2^3, 2^3);

julia> itensor(m, s', dag(s))
ITensor ord=6 (dim=2|id=535|"Qubit,Site,n=1")' (dim=2|id=101|"Qubit,Site,n=2")' (dim=2|id=268|"Qubit,Site,n=3")' (dim=2|id=535|"Qubit,Site,n=1") (dim=2|id=101|"Qubit,Site,n=2") (dim=2|id=268|"Qubit,Site,n=3")
ITensors.NDTensors.Dense{Float64, Vector{Float64}}
```

Reply:
Gotcha.